trieTools/trieToolsCommon/addressTokens.go

```diff
@@ -0,0 +1,103 @@
+package trieToolsCommon
+
+// This is not concurrent safe
+type addressTokensMap struct {
+	internalMap map[string]map[string]struct{}
+}
+
+func NewAddressTokensMap() AddressTokensMap {
+	return &addressTokensMap{
+		internalMap: make(map[string]map[string]struct{}),
+	}
+}
+
+// Add will add all provided tokens to the corresponding address
+func (atm *addressTokensMap) Add(address string, tokens map[string]struct{}) {
+	_, addressExists := atm.internalMap[address]
+	if !addressExists {
+		atm.internalMap[address] = make(map[string]struct{})
+	}
+
+	atm.addTokens(address, tokens)
+}
+
+func (atm *addressTokensMap) addTokens(address string, tokens map[string]struct{}) {
+	tokensCopy := copyTokens(tokens)
+	for token := range tokensCopy {
+		atm.internalMap[address][token] = struct{}{}
+	}
+}
+
+func copyTokens(tokens map[string]struct{}) map[string]struct{} {
+	ret := make(map[string]struct{})
+	for token := range tokens {
+		ret[token] = struct{}{}
+	}
+
+	return ret
+}
+
+// HasAddress checks if the address is in map
+func (atm *addressTokensMap) HasAddress(address string) bool {
+	_, found := atm.internalMap[address]
+	return found
+}
+
+// NumAddresses returns the num of addresses in map
+func (atm *addressTokensMap) NumAddresses() uint64 {
+	return uint64(len(atm.internalMap))
+}
+
+// NumTokens returns the num of tokens in map for all addresses
+func (atm *addressTokensMap) NumTokens() uint64 {
+	numTokens := uint64(0)
+	for _, tokens := range atm.internalMap {
```

Comment:
... or could have been a separate counter stored in the addressTokensMap and increased whenever you add a new map of tokens

Reply:
This func is only created to show statistics 3 times(when reading input file) in debug and is pretty efficient anyway.
IMHO, I would not add a counter which I have to maintain/sync after every operation (add, delete) which adds extra logic and will be called more than 3 times

Reply:
ok